ls1intum/Artemis
# `General`: Add time zone information to date time picker

### Checklist
#### General


#### Client

### Motivation and Context


While setting the release date for a course/exercise/lecture you are unable to see the timezone of the date you are setting. It seems like that Artemis is always taking the local timezone of the user. If you are for example in an other timezone than your students, you might get offsets for the dates.

For example if you are in EEST (UTC +3) and set the release time for 2021-09-01T08:00:00 the date which will be shown to students in Munich (CEST (UTC +2)) is 2021-09-01T07:00:00.

Closes https://github.com/ls1intum/Artemis/issues/3966

### Description

Added an optional time zone warning to the date picker component and used it in course, lecture and exercise creation.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a course
4. Check that the time zone warning is correct and displayed
5. Switch time zone on your computer
6. Check the time zone again
7. Repeat for lecture and exercises

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots



<img width="524" alt="Bildschirmfoto 2021-10-24 um 14 18 22" src="https://user-images.githubusercontent.com/51987021/138594410-74076b32-3acf-49b1-8a12-8e91dba8455b.png">
<img width="460" alt="Bildschirmfoto 2021-10-24 um 14 18 16" src="https://user-images.githubusercontent.com/51987021/138594414-7faeff19-ee2e-428f-bd3f-eae16375119c.png">
